src/ppca.jl

```diff
@@ -0,0 +1,197 @@
+# Probabilistic Principal Component Analysis
+
+"""Probabilistic PCA type"""
+type PPCA{T<:AbstractFloat}
+    mean::Vector{T}       # sample mean: of length d (mean can be empty, which indicates zero mean)
+    W::Matrix{T}          # weight matrix: of size d x p
+    σ²::T                 # residual variance
+
+    function (::Type{PPCA}){T}(mean::Vector{T}, W::Matrix{T}, var::T)
+        d = size(W, 1)
+        (isempty(mean) || length(mean) == d) ||
+            throw(DimensionMismatch("Dimensions of weight matrix and mean are inconsistent."))
+        new{T}(mean, W, var)
+    end
+end
+
+## properties
+
+MultivariateStats.indim(M::PPCA) = size(M.W, 1)
```

Comment:
You don't need to prefix with the module name to overload functions defined in the module.